addons/mail/controllers/thread.py

```diff
@@ -230,7 +234,12 @@ def mail_message_update_content(self, message_id, body, attachment_ids, attachme
                 if key in thread._get_allowed_message_update_params()
             }
         )
-        return Store(message).get_result()
+        store_data = Store(message).get_result()
+        target = guest or request.env.user
+        target._bus_send("mail.record/insert", store_data)
+        if self.env.context.get('bus_rpc_uuid'):
+            target._bus_send("bus.rpc/end", self.env.context.get('bus_rpc_uuid'))
```

Comment:
I’m not sure which of the two implementations we want to keep, but there currently are two different implementations coexisting: the one using the decorator and the one using the context. We probably shouldn’t propagate this context anyway since we only use it in the top-level controller function.